asset-transfer-basic/chaincode-go/assetTransfer.go

```diff
@@ -0,0 +1,179 @@
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// SmartContract provides functions for managing an Asset
+type SmartContract struct {
+	contractapi.Contract
+}
+
+// Asset describes basic details of what makes up a simple asset
+type Asset struct {
+	ID             string `json:"ID"`
+	Color          string `json:"color"`
+	Size           string `json:"size"`
+	Owner          string `json:"owner"`
+	AppraisedValue string `json:"appraisedValue"`
+}
+
+// QueryResult structure used for handling result of query
+type QueryResult struct {
+	Key    string `json:"Key"`
+	Record *Asset
+}
+
+// InitLedger adds a base set of assets to the ledger
+func (s *SmartContract) InitLedger(ctx contractapi.TransactionContextInterface) error {
+	assets := []Asset{
+		Asset{ID: "asset1", Color: "blue", Size: "small", Owner: "Tomoko", AppraisedValue: "3"},
+		Asset{ID: "asset2", Color: "red", Size: "small", Owner: "Brad", AppraisedValue: "4"},
+		Asset{ID: "asset3", Color: "green", Size: "medium", Owner: "Jin Soo", AppraisedValue: "5"},
+		Asset{ID: "asset4", Color: "yellow", Size: "medium", Owner: "Max", AppraisedValue: "6"},
+		Asset{ID: "asset5", Color: "black", Size: "large", Owner: "Adriana", AppraisedValue: "7"},
+		Asset{ID: "asset6", Color: "white", Size: "large", Owner: "Michel", AppraisedValue: "8"},
+	}
+
+	for i, asset := range assets {
+		assetAsBytes, _ := json.Marshal(asset)
+		err := ctx.GetStub().PutState("ASSET"+strconv.Itoa(i), assetAsBytes)
+
+		if err != nil {
+			return fmt.Errorf("Failed to put to world state. %s", err.Error())
+		}
+	}
+
+	return nil
+}
+
+// CreateAsset issues a new asset to the world state with given details.
+func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface, assetID, id, color, size, owner, appraisedValue string) error {
+	asset := Asset{
+		ID:             id,
+		Color:          color,
+		Size:           size,
+		Owner:          owner,
+		AppraisedValue: appraisedValue,
+	}
+
+	assetAsBytes, _ := json.Marshal(asset)
+
+	return ctx.GetStub().PutState(assetID, assetAsBytes)
+}
+
+// ReadAsset returns the asset stored in the world state with given id.
+func (s *SmartContract) ReadAsset(ctx contractapi.TransactionContextInterface, assetID string) (*Asset, error) {
+	assetAsBytes, err := ctx.GetStub().GetState(assetID)
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read from world state. %s", err.Error())
+	}
+	if assetAsBytes == nil {
+		return nil, fmt.Errorf("%s does not exist", assetID)
+	}
+
+	asset := new(Asset)
+	_ = json.Unmarshal(assetAsBytes, asset)
+
+	return asset, nil
+}
+
+// UpdateAsset updates an existing asset in the world state with provided parameters.
+func (s *SmartContract) UpdateAsset(ctx contractapi.TransactionContextInterface, assetID, id, color, size, owner, appraisedValue string) error {
+	assetAsBytes, err := ctx.GetStub().GetState(assetID)
+
```

Comment:
AssetExists() can be used (ultimately does the same thing, but may be better to show the re-use).

Reply:
done.
After I looked into it a bit more, IMO this is a bit redundant, `AssetExists()` probably should return assetJSON if asset exist, and return nil if not. In this case, it will share smilar functionality as `ReadAsset()`. Therefore `AssetExists()` might not be necessary anymore. However, in order to keep in sync with the generated code, I'm cool with having them both.